storage/mysql/tree_storage.go

```diff
@@ -404,6 +404,13 @@ func (t *treeTX) Rollback() error {
 	return nil
 }
 
+func (t *treeTX) Close() error {
+	if !t.closed {
+		return t.Rollback()
```

Comment:
As were most tx.Rollback() calls before it.

From a theoretical standpoint, I think Close() returning an error is the proper way to do it. I could add logging to Close, if you prefer, but I think errors arising from it are uninteresting: if Close() is taking action, that's because something else already went wrong, and that something else is the error we really care about.